File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
    let stack = [];
    
    for (const word of s) {
        if (stack.length === 0) {
            stack.push(word)
        }
        else {
            if (stack[stack.length - 1] === word) {
                stack.pop();
            } else {
                stack.push(word);
            }
        }
    }
    
    return stack.length ? 0 : 1
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    arr = []
    
    for i in s:
        if len(arr) == 0:
            arr.append(i)
        else:
            if arr[len(arr) - 1] != i:
                arr.append(i)
            else:
                arr.pop()
                
    answer = 0
    if len(arr) == 0:
        answer = 1
    
    return answer